### PR TITLE
Updated logic to replace space with non breaking space

### LINKED ISSUE #155: Image upload issues in controlled editor
- Uploaded images are not displayed in the controlled editor.

bug _t, @AbhayVAshokan _a

### PR DESCRIPTION
**Pull request by AbhayVAshokan**
Fixes #155 
@labeebklatif _a please review.
Explanation: https://vimeo.com/672342688/2aac06adc1

cc: @vysakh-poulose

### CODE REVIEW

File: lib/components/Editor/index.js [hunk 1]
```diff
@@ -74,8 +74,16 @@ const Tiptap = (
   const addonOptions = addons.map((option) => option.toLowerCase());
   isUnsplashImageUploadActive && addonOptions.push("image-upload");
   const allOptions = defaultOptions.concat(addonOptions);
-  const formattedValue =
-    typeof value === "string" ? value.replaceAll(" ", "&nbsp;") : "";
+
+  // Replace all spaces within tags with "nbsp;". This prevents tiptap editor from removing trailing spaces while typing.
+  let formattedValue = "";
```
Comment (labeebklatif) [inline lib/components/Editor/index.js:79]: @AbhayVAshokan Can we move this to a util function. `index` is already too crowded

**Review by labeebklatif — COMMENTED** [review]
@AbhayVAshokan _a Please have a look

**Comment by AbhayVAshokan** [comment]
@labeebklatif _a please review the changes.

**Review by labeebklatif — APPROVED** [review]
LGTM!

**Pull request merged**